libraries/AP_HAL_ChibiOS/system.cpp

```diff
@@ -259,7 +259,12 @@ void panic(const char *errormsg, ...)
 
 uint32_t micros()
 {
+#if CH_CFG_ST_RESOLUTION == 32 && CH_CFG_ST_FREQUENCY==1000000U
+    // special case optimisation for 32 bit timers
+    return st_lld_get_counter();
+#else
```

Comment:
can we add a #warn or #error for cases where CH_CFG_ST_RESOLUTION == 32 and CH_CFG_ST_FREQUENCY is not set to 1000000U, by mistake or something.

Reply:
some boards (AP_Periph f1 for example) deliberately don't set to 1000000 as it costs too much cpu